### ISSUE TITLE
[BUG] cannot get version number

### ISSUE THREAD

**Issue by bbuzzanga** [aria-tools/ARIA-tools#412]
I'm running AT on a set of older products (<=v.2.05)
 ariaTSsetup crashes on command:
`ariaTSsetup.py -f '/u/leffe-data2/buzzanga/data/VLM/Sentinel1/HR_ARIA_2024a/products/S1*.nc' -b '36.75 37.225 -76.655 -75.928' -w /u/leffe-data2/buzzanga/data/VLM/Sentinel1/HR_ARIA_2024a -mo 0.0081 -d Download -m Download -nt 16`

with error:
```Traceback (most recent call last):
  File "/u/leffe-data2/buzzanga/Miniconda3/envs/ARIA/bin/ariaTSsetup.py", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/u/leffe-data2/buzzanga/Software_InSAR/ARIA-tools_git/tools/bin/ariaTSsetup.py", line 632, in <module>
    main()
  File "/u/leffe-data2/buzzanga/Software_InSAR/ARIA-tools_git/tools/bin/ariaTSsetup.py", line 410, in main
    standardproduct_info = ARIAtools.product.Product(
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/u/leffe-data2/buzzanga/Software_InSAR/ARIA-tools_git/tools/ARIAtools/product.py", line 420, in __init__
    self.__run__()
  File "/u/leffe-data2/buzzanga/Software_InSAR/ARIA-tools_git/tools/ARIAtools/product.py", line 1247, in __run__
    self.products = self.__continuous_time__()
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/u/leffe-data2/buzzanga/Software_InSAR/ARIA-tools_git/tools/ARIAtools/product.py", line 1009, in __continuous_time__
    vers.append(ver_num)
                ^^^^^^^
UnboundLocalError: cannot access local variable 'ver_num' where it is not associated with a value
```

**Comment by alexfore** [comment]
To replicate this issue I need more than has been provided in this bug report -- where can the input products for this be obtained or how may they be obtained using ariaDownload?

**Comment by alexfore** [comment]
The issue is in these lines of code
```
                    unw_f = sc[1]['unwrappedPhase']
                    basename_unw = os.path.basename(
                        unw_f.split('"')[1])
                    # parse version from NISAR and S1 GUNWs appropriately
                    ext = os.path.splitext(unw_f)[1].lower()
                    if ext == '.nc':
                        ver_str = re.search(
                            r'(v\d+_\d+_\d+.*)\.',
                            basename_unw).group(1)
                        ver_num = float(ver_str[1:].replace('_', ''))
                    if ext == '.h5':
                        version = unw_f.split('_')[-1][:-3]
                        version = float(version)
                    vers.append(ver_num)
```
I see a lot of issues here:
* should be using `if` / `elif` / `else` log error and raise exception if fall through the prior two conditionals) and not two `if`s since we cannot tell if we have fallen through with two `if`s
* the block of code if the second conditional is `True` seems wrong (should be ver_num instead of version)
* shouldn't we be using basename_unw to get the ext from and not unw_f?
@sssangha can you comment on these and let me know if you agree? Thanks!

**Comment by alexfore** [comment]
Generally I think the products data structure here is way too complex, IMO it would be beneficial to define our own class for that, perhaps subclassing dict or namedtuple + some syntax suger methods so the interface is simple and expressive instead of this

**Comment by alexfore** [comment]
I've implemented what I mentioned above here, LMK is this is ok or not: https://github.com/aria-tools/ARIA-tools/compare/issue_412?expand=1

**Comment by sssangha** [comment]
@alexfore , I tested the proposed patch, and it will handle NISAR GUNW products with just one tweak. Not sure of the most appropriate way to push it over your proposed commit so I'll outline my fix here.

This [line](https://github.com/aria-tools/ARIA-tools/blob/bc7f0a31907dc4350aa45feb8f3f1b60728b0df0/tools/ARIAtools/product.py#L1007), should be:
`ver_str = basename_unw.split('_')[-1][:-3]`

The variable `basename_unw` (e.g. `NISAR_L2_PR_GUNW_001_001_A_001_001_4000_HHNA_20100410T062238_20100410T062252_20110111T061858_20110111T061912_D00401_P_F_J_001.h5`) needs to be referenced for the version string to be isolated with `.split('_')[-1][:-3]` (e.g. `ver_num = float(001)`). As is, the program crashes because `unw_f` (e.g. `NETCDF:"/u/trappist-r0/ssangha/ARIA_tests/NISAR/products/NISAR_L2_PR_GUNW_001_001_A_001_001_4000_HHNA_20100410T062238_20100410T062252_20110111T061858_20110111T061912_D00401_P_F_J_001.h5":/science/LSAR/GUNW/grids/frequencyA/unwrappedInterferogram/HH/unwrappedPhase`) is specified, such that `ver_num = float('001.h5":/science/LSAR/GUNW/grids/frequencyA/unwrappedInterferogram/HH/unwrappedPh')`.

This would also be consistent with the `if` statement above where the `basename_unw` is  used.

Please let me know if I should clarify anything here. Thanks

**Comment by sssangha** [comment]
> Generally I think the products data structure here is way too complex, IMO it would be beneficial to define our own class for that, perhaps subclassing dict or namedtuple + some syntax suger methods so the interface is simple and expressive instead of this

Agreed, this part of the code is very complex. I would be glad to outline a streamlined way to simplify it as you suggest.

**Comment by alexfore** [comment]
closed by 3ced73b68eec00b787300a61d65c5b60a926808b
